src/components/Inbox/MessageCard.tsx

```diff
@@ -14,41 +14,62 @@ import { UpdateMessageIsReadDocument } from "src/graphql/mutations/UpdateMessage
 import { GetConnectionMessagesQuery } from "src/graphql/queries/GetConnectionMessages.generated";
 import {
   formatDate,
-  getFullName,
   getMessageSenderType,
-  getMessageAvailabilityLabel,
 } from "src/utils";
-import { STRIPPED_MESSAGE } from "src/utils/constants";
-import { TConnectionListItem } from "./ConnectionListSidebar/ConnectionListItem";
 
 export type TMessageItem =
   GetConnectionMessagesQuery["connection"]["messages"]["edges"][0]["node"];
 interface Props {
   message: TMessageItem;
-  connection: TConnectionListItem;
   className?: string;
 }
 
 const MessageCard: React.FC<Props> = ({
   message,
-  connection,
   className,
 }: Props) => {
   const senderType = getMessageSenderType(message);
-  const senderName =
-    senderType === "visitor" ? getFullName(connection.visitor) : "Me";
-  const MessageAvailabilityIcon = () => {
+  const MessageErrorIcon = () => (
+    message.availability === MessageAvailability.Rejected ?
+      <StopOutlined className="text-red-500" /> :
+      null
+  )
+  const MessageDeliveryIcon = () => {
     switch (message.availability) {
+      case MessageAvailability.Available:
+        return senderType === "inmate" ? <CheckOutlined className="text-blue-500" /> : null;
       case MessageAvailability.Pending:
         return <ClockCircleOutlined className="text-gray-500" />;
       case MessageAvailability.Rejected:
-        return <StopOutlined className="text-red-500" />;
+        return null;
+      default:
+        assertNever(message.availability);
+    }
+  };
+  const MessageContent = () => {
+    switch (message.availability) {
       case MessageAvailability.Available:
-        return <CheckOutlined className="text-blue-500" />;
+        return <span>{message.content}</span>;
+      case MessageAvailability.Pending:
+        return senderType === "inmate" ? <span>{message.content}</span> : <span className="text-gray-400">This message is hidden to you while pending review.</span>;
+      case MessageAvailability.Rejected:
+        return senderType === "inmate" ? <span>{message.content}</span> : <span className="text-gray-400">This message is unavailable because it was rejected.</span>;
       default:
         assertNever(message.availability);
     }
   };
+  const messageAvailabilityText = (() => {
```

Comment:
building a string const this way with an inline function call seems... better than using a nested ternary, I suppose. I guess I don't like either way. The ternary would be
```suggestion
const messageAvailabilityText = (
  message.availability === MessageAvailability.Available && senderType === "inmate" ?
     "Delivered" : 
     message.availability === MessageAvailability.Pending ?
      "Review Required" : 
      message.availability === MessageAvailability.Rejected ?
        "Rejected" : 
        null
);
```

Reply:
def not nested a ternary! it's v hard to read

Reply:
i think this anonymous function implementation works well!